skills_ml/algorithms/embedding/base.py

```diff
@@ -17,115 +18,114 @@ class Word2VecModel(object):
 
     Example:
     ```
-    from airflow.hooks import S3Hook
     from skills_ml.algorithms.embedding.base import Word2VecModel
 
-    s3_conn = S3Hook().get_conn()
-    word2vec_model = Word2VecModel(s3_conn=s3_conn)
+    word2vec_model = Word2VecModel()
+    word2vec_model.load_model(path)
     ```
     """
-    def __init__(self, model_name='word2vec_gensim_2017-07-14T11:32:00.997426',
-        model=None, s3_conn=None, s3_path=S3_PATH_EMBEDDING_MODEL):
+    def __init__(self, storage=FSStore(), model=None):
         """
         Attributes:
-            model_name (str): name of the model to be used.
-            s3_path (str): the path of the model on S3.
-            s3_conn (:obj: `boto.s3.connection.S3Connection`): the boto object to connect to S3.
-            files (:obj: `list` of (str)): model files need to be downloaded/loaded.
+            storage (:obj: `skills_ml.Store`): skills_ml Store object
             model (:obj: `gensim.models.doc2vec.Doc2Vec`): gensim doc2vec model.
-            lookup (dict): lookup table for mapping each jobposting index to soc code.
         """
+        self.storage = storage
+        self._model = model
+        self.model_name = None
+
+    @property
+    def storage(self):
+        return self._storage
+
+    @storage.setter
+    def storage(self, value):
+        if value.__class__.__name__ in [c.__name__ for c in Store.__subclasses__()]:
```

Comment:
I think using duck typing could work better here. Part of the appeal here is that a user can simply write their own classes, that may or may not be subclasses of our Store class. We could just make sure that it has .write and .load methods.